lib/core/src/sdk.rs

```diff
@@ -3920,7 +4371,44 @@ impl LiquidSdk {
     /// the new correct `webhook_url`. To unregister a webhook call [LiquidSdk::unregister_webhook].
     pub async fn register_webhook(&self, webhook_url: String) -> SdkResult<()> {
         info!("Registering for webhook notifications");
-        self.persister.set_webhook_url(webhook_url)?;
+        let maybe_old_webhook_url = self.persister.get_webhook_url()?;
+
+        self.persister.set_webhook_url(webhook_url.clone())?;
+
+        // If the webhook URL has changed, update all bolt12 offers
+        // that were created with the old webhook URL or without a webhook URL
+        if let Some(old_webhook_url) = maybe_old_webhook_url {
+            if old_webhook_url != webhook_url {
```

Comment:
Don't we need to go over the offers regardless if the url has changed? I am thinking of another device that has created the offer with a different hook url than ours.

Reply:
Are you saying we should update the offer webhook URL regardless of which device created it / set the webhook URL?

Reply:
Currently IIUC multiple devices use only one offer (a device doesn't create an offer if one already exists).
In that case it makes sense to update by the last online device otherwise we may loose track of it.

Reply:
Done [3773775](https://github.com/breez/breez-sdk-liquid/pull/882/commits/3773775d328ed481a92c2d1450c0a0a6800a3142)